src/parameters.ts

```diff
@@ -1,82 +1,82 @@
-import * as CryptoJS from "crypto-js";
-
 import { hashBitCount } from "./utils";
+import * as crypto from "crypto";
+
+export interface PrimeGroup {
+  N: bigint; // the prime
+  g: bigint; // a generator of the multiplicative group Zn
+}
 
-type HashingAlgorithm = keyof typeof SRPParameters.H;
+export type HashFunction = (data: ArrayBuffer) => Promise<ArrayBuffer>;
 
 export class SRPParameters {
-  public static N: {
-    "256": bigint;
-    "512": bigint;
-    "768": bigint;
-    "1024": bigint;
-    "1536": bigint;
-    "2048": bigint;
-  };
-
-  public static g: bigint;
+  public static PrimeGroup: { [key: number]: PrimeGroup };
 
-  public static H: {
-    SHA1: HashingAlgorithm;
-    SHA224: HashingAlgorithm;
-    SHA256: HashingAlgorithm;
-    SHA384: HashingAlgorithm;
-    SHA512: HashingAlgorithm;
-    SHA3: HashingAlgorithm;
-    RIPEMD160: HashingAlgorithm;
-  };
+  public static H: { [key: string]: HashFunction };
 
   public readonly NBits: number;
-  public readonly H: any;
   public readonly HBits: number;
 
   constructor(
-    public readonly N: bigint = SRPParameters.N[2048],
-    public readonly g: bigint = SRPParameters.g,
-    H: HashingAlgorithm = SRPParameters.H.SHA512,
+    public readonly primeGroup: PrimeGroup = SRPParameters.PrimeGroup[2048],
+    public readonly H: HashFunction = SRPParameters.H.SHA512,
   ) {
-    this.NBits = this.N.toString(2).length;
+    this.NBits = this.primeGroup.N.toString(2).length;
 
-    const hasher = CryptoJS.algo[H];
-
-    if (!hasher || !(hasher instanceof (CryptoJS.lib as any).Hasher.init)) {
-      throw new Error("Unknown hash function");
+    if (!H) {
+      throw new Error("Hash function required");
     }
 
-    this.H = hasher.create();
     // Calculate size of hash output
-    this.HBits = hashBitCount(this);
+    this.HBits = hashBitCount(this); // TODO fix this, it will have to be async
```

Comment:
This could be a problem. The async nature of webcrypto's digest function is good for performance but creates difficulty of calculating the bit count in the constructor

Reply:
I just saw that WebCrypto is async right now. My gut reaction was negative, that a sync api is always nicer than an async one, and it would be sad to turn tssrp6a into a Promise fest.
But for the final user that runs this in his browser, sync is not great. It might make the UI unresponsive during hashing operations.

I think you can just move HBits out of the constructor here. It's only used by `generateRandomSalt`, from `createVerifierAndSalt`, so it's fine if this moved completely out of `SRPParameters`.